mmv1/templates/terraform/examples/spanner_instance_partition_basic.tf.tmpl

```diff
@@ -0,0 +1,14 @@
+resource "google_spanner_instance" "main" {
+  name         = "instance-${local.name_suffix}"
+  config       = "regional-us-central1"
+  display_name = "main-instance"
+  num_nodes    = 1
+}
+
+resource "google_spanner_instance_partition" "partition" {
+  name         = "partition-${local.name_suffix}"
```

Comment:
The instance partition name should be provided as a `var` in the `example` in InstancePartition.yaml file.
Example: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/spanner/InstanceConfig.yaml#L55